i6_models/assemblies/conformer/conformer.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+import torch
+from torch import nn
+
+# TODO probably used for SpecAugment
+from torchaudio.transforms import TimeMasking, TimeStretch, FrequencyMasking
+from dataclasses import dataclass
+
+from i6_models.config import ModelConfiguration
+
+from i6_models.parts.conformer.convolution import ConformerConvolutionV1
+from i6_models.parts.conformer.feedforward import ConformerPositionwiseFeedForwardV1
+from i6_models.parts.conformer.mhsa import ConformerMHSAV1
+
+
+@dataclass
+class ConformerFeedForwardV1Config(ModelConfiguration):
+    dim: int
+
+
+@dataclass
+class ConformerMHSAV1Config(ModelConfiguration):
+    pass
+
+
+@dataclass
+class ConformerConvV1Config(ModelConfiguration):
+    pass
+
+
+@dataclass
+class ConformerBlockV1Config(ModelConfiguration):
+    # hyperparameters
+    norm_dim: int
+
+    # nested configurations
+    ff_cfg: ConformerFeedForwardV1Config  # Configuration for ConformerPositionwiseFeedForwardV1
+    mhsa_cfg: ConformerMHSAV1Config  # Configuration for ConformerMHSAV1
+    conv_cfg: ConformerConvV1Config  # Configuration for ConformerConvolutionV1
+
+
+@dataclass
+class ConformerFrontendV1Config(ModelConfiguration):
+
+    feature_dim: int
+    hidden_dim: int
+    dropout: float
+
+    # TODO: Maybe put this in own config?
+    conv_stride: int
+    conv_kernel: int
+    conv_stride: int
+    conv_padding: int
+
+    spec_aug_cfg: ModelConfiguration  # TODO
+
+
+@dataclass
+class ConformerV1Config(ModelConfiguration):
+    # hyperparameters
+    num_layers: int
+
+    # nested configurations
+    front_cfg: ConformerFrontendV1Config
+    block_cfg: ConformerBlockV1Config
+
+
+class ConformerBlockV1(nn.Module):
+    """
+    Conformer block module
+    """
+
+    def __init__(self, cfg: ConformerBlockV1Config):
+        """
+        :param cfg: Conformer Block Configuration with subunits for the different Conformer parts
+        """
+        super().__init__()
+
+        self.ff_config = cfg.ff_cfg
+        self.mhsa_config = cfg.mhsa_cfg
+        self.conv_config = cfg.conv_cfg
+
+        self.ff_1 = ConformerPositionwiseFeedForwardV1(cfg=self.ff_config)
+        self.mhsa = ConformerMHSAV1(cfg=self.mhsa_config)
+        self.conv = ConformerConvolutionV1(cfg=self.conv_config)
+        self.ff_2 = ConformerPositionwiseFeedForwardV1(cfg=self.ff_config)
+        self.final_layer_norm = torch.nn.LayerNorm(self.ff_config.dim)
+
+    def forward(self, tensor: torch.Tensor):
+        """
+        :param tensor: input tensor of shape [B, T, F]
+        :return: torch.Tensor of shape [B, T, F]
+        """
+        residual = tensor  #  [B, T, F]
+        x = self.ff_1(tensor)  #  [B, T, F]
```

Comment:
The feature dim might change here, right? Should we account for that in the commented shape or do we only care about the order of axes?

Reply:
IMO this should be indicated in the docstring, e.g. input `[B, T, F]`, output `[B, T, F']`.

Reply:
According to doc of the FF module it should not change.